openwisp_controller/config/static/config/js/relevant_templates.js

```diff
@@ -0,0 +1,116 @@
+            }
+            $('.sortedm2m-container > .help').text(helpText);
+        },
+        addChangeEventHandlerToBackendField = function () {
+            $('#id_config-0-backend').change(function () {
+                setTimeout(function () {
+                    // ensures getDefaultTemplates execute only after other
+                    // onChange event handlers attached this field has been
+                    // executed.
+                    showRelevantTemplates();
+                });
+            });
+            // Initially request data to populate everything
+            showRelevantTemplates();
+        },
+        updateConfigTemplateField = function (templates) {
+            $('input[name="config-0-templates"]').attr(
+                'value', templates.join(',')
+            );
+            $('input.sortedm2m:first').trigger('change');
+        },
+        showRelevantTemplates = function () {
+            var orgID = $('#id_organization').val(),
+                backend = $('#id_config-0-backend').val();
+
+            // Hide templates if no organization or backend is selected
+            if (orgID.length === 0 || backend.length === 0) {
+                resetTemplateOptions();
+                updateTemplateHelpText();
+                return;
+            }
+            var url = window._relevantTemplateUrl.replace('org_id', orgID);
+            // Get relevant templates of selected org and backend
+            url = url + '?backend=' + backend;
+            $.get(url).done(function (data) {
+                resetTemplateOptions();
+                var enabledTemplates = [];
+                // Loop over all input elements fo  the templates field.
+                // If value(templateId) of an element is not in data, then
+                //      uncheck, enable and hide the element
+                // else.
+                //      check, the element and disable if if a required template
+                Object.keys(data).map(function (templateId, index){
+                    var element = getTemplateOptionElement(index, templateId, data[templateId].name),
+                        prefixElement = getTemplateOptionElement(index, templateId, data[templateId].name, true),
+                        inputField = element.children().children('input');
+
+                    if (data[templateId].default && (isFirstRun() !== true)) {
+                        inputField.prop('checked', true);
+                        enabledTemplates.push(templateId);
+                    }
+
+                    if (data[templateId].required) {
+                        inputField.prop('disabled', true);
+                        inputField.prop('checked', true);
+                        enabledTemplates.push(templateId);
+                    }
+
+                    $('fieldset ul.sortedm2m-items:visible').append(element);
+                    $('fieldset ul.sortedm2m-items:not(:visible)').append(prefixElement);
```

Comment:
Prefix elements `__prefix__` are used by Django internally for form validation. It was working fine without adding them here, but I insisted on adding them to avoid any future bugs.